server/game/cards/01_SOR/units/FirstLegionSnowtrooper.ts

```diff
@@ -0,0 +1,22 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { KeywordName } from '../../../core/Constants';
+
+export default class FirstLegionSnowtrooper extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '4619930426',
+            internalName: 'first-legion-snowtrooper',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addConstantAbility({
+            title: 'While attacking a damaged unit, this unit gets +2/+0 and gains Overwhelm. (Deal excess damage to the opponent\'s base.)',
+            condition: (context) => context.source.isAttacking() && context.source.activeAttack?.target.isUnit() && (context.source.activeAttack?.target.location !== 'discard' && context.source.activeAttack?.target.damage > 0),
```

Comment:
There's almost nowhere we need to use raw strings, we have enums that cover pretty much everything. In this case should use `Location.Discard` instead of `'discard'`

Reply:
Although actually I think this check might not be fully correct, because it's possible that the card ended up somewhere other than discard (e.g., some hypothetical effect could bounce it back to hand). So it should be like this:

```typescript
import * as EnumHelpers from '../../../core/utils/EnumHelpers';

// inside card impl

condition: (context) => context.source.isAttacking() && context.source.activeAttack?.target.isUnit() && (EnumHelpers.isArena(context.source.activeAttack?.target.location) && context.source.activeAttack?.target.damage > 0),
```

Reply:
thank you for the comment, i'll be more vigilant in searching for a more appropriate enumerator.